integration/tests/cook/test_cli.py

```diff
@@ -2146,9 +2146,18 @@ def test_avoid_exit_on_connection_error(self):
             self.assertIn('Encountered connection error with bar', cli.decode(cp.stderr))
 
     def test_submit_with_gpus(self):
+        settings_dict = util.settings(self.cook_url)
+        gpu_enabled = settings_dict['mesos-gpu-enabled']
+        valid_gpu_models_config_map = settings_dict.get("pools", {}).get("valid-gpu-models", [])
         cp, uuids = cli.submit('ls', self.cook_url, submit_flags=f'--gpus 1')
-        if util.settings(self.cook_url)['mesos-gpu-enabled']:
-            self.assertEqual(0, cp.returncode, cp.stderr)
+        if gpu_enabled:
+            if valid_gpu_models_config_map:
```

Comment:
Don't we nee to also check that the pool we're submitting to has a matching pool regex in the config? If we're submitting to `k8s-spot` but only have `k8s-ondemand` in the `valid-gpu-models`, then I think this will fail, right?